google-analytics-dashboard.html

```diff
@@ -32,66 +32,97 @@
 @homepage https://googlewebcomponents.github.io/google-analytics
 -->
 
-<polymer-element
-  name="google-analytics-dashboard"
-  extends="google-analytics-base"
-  on-analytics-dashboard-control-change="{{queryUpdated}}">
 
+<dom-module id="google-analytics-dashboard">
   <template>
-    <content></content>
+    <google-signin-aware
+      on-google-signin-aware-success="signedIn"
+      on-google-signin-aware-signed-out="signedOut"></google-signin-aware>
+    <content id="content"></content>
   </template>
-  <script>
-
-    Polymer('google-analytics-dashboard', {
-
-      ready: function() {
-        /**
-         * The `query` attribute represents the internal query object of this
-         * dashboard. It is updated when control elements fire the
-         * `analytics-dashboard-control-change` event and pass along query data.
-         *
-         * @attribute query
-         * @type object
-         */
-        this.query = {};
-
-        this.updateChildren();
-      },
+</dom-module>
+<script>
 
-      /**
-       * The `queryUpdated` method is the callback for whenever the
-       * `analytics-dashboard-control-change` event is fired. It updates the
-       * query attribute, which is then sent to child charts.
-       *
-       * @method queryUpdated
-       * @param {CustomEvent} event - The event with the query data.
-       */
-      queryUpdated: function(event) {
-        // Update `this.query` with the passed event data.
-        Object.keys(event.detail).forEach(function(key) {
-          this.query[key] = event.detail[key];
-        }.bind(this))
+  'use strict';
```

Comment:
You mean, you want every Polymer inside IIFE. Not a standard practice, but
I'll do it.

On Mon, May 4, 2015 at 3:58 PM, Philip Walton <notifications@github.com>
wrote:

> In google-analytics-dashboard.html
> <https://github.com/GoogleWebComponents/google-analytics/pull/24#discussion_r29632127>
> :
>
> >
> > -      /**
> > -       * The `queryUpdated` method is the callback for whenever the
> > -       * `analytics-dashboard-control-change` event is fired. It updates the
> > -       * query attribute, which is then sent to child charts.
> > -       *
> > -       * @method queryUpdated
> > -       * @param {CustomEvent} event - The event with the query data.
> > -       */
> > -      queryUpdated: function(event) {
> > -        // Update `this.query` with the passed event data.
> > -        Object.keys(event.detail).forEach(function(key) {
> > -          this.query[key] = event.detail[key];
> > -        }.bind(this))
> > +  'use strict';
>
> Needs an IIFE. This 'use strict' is in the global scope.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleWebComponents/google-analytics/pull/24/files#r29632127>
> .
>